### PR TITLE
Bypass

### PR DESCRIPTION
**Pull request by rssk**
# problem statement
- There is no way to call a protected method on a shielded model. This is a problem when using shielded models for a context with no user.
# solution
- Add a bypass method that can call a protected method.

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "requireSpacesInAnonymousFunctionExpression": {
         "beforeOpeningCurlyBrace": true
     },
+    "requirePaddingNewLinesAfterBlocks": false,
```
Comment (rssk) [inline .jscsrc:11]: I think this was added by coins-validate (older version)? Anyway, BK-shield needs to be converted to eslint, I figured this pull was not the place to do so.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# problem statement
+- What does is the problem or feature request.
```
Comment (cdaringe) [inline .github/PULL_REQUEST_TEMPLATE.md]: typo

File: .jscsrc (hunk 1 above)
Comment (cdaringe) [inline .jscsrc:11]: 👍

File: docs/wrapRelations.js.html [hunk 3]
```diff
@@ -73,13 +73,13 @@ <h1 class="page-title">Source: wrapRelations.js</h1>
...
 </nav>
 
 <br class="clear">
 
 <footer>
-    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.4.0</a> on Mon Feb 08 2016 12:38:09 GMT-0700 (MST)
+    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.4.0</a> on Fri May 06 2016 15:18:15 GMT-0600 (MDT)
```
Comment (cdaringe) [inline docs/wrapRelations.js.html]: please `git rm -rf docs`.  coins-validate will handle this automatically later

File: lib/secureAccessMethods.js [hunk 4]
```diff
@@ -174,10 +174,28 @@ function createSecure(user, options) {
         });
 }
 
+/**
+ * allows a protected method to be called
+ * @param  {string} method   protected method to call
+ * @param  {object} options  options to call function with
+ * @return {Promise}         Promise that the called function returns
```
Comment (cdaringe) [inline lib/secureAccessMethods.js:181]: cory recently pointed out that `return` actually isnt a thing.  `returns` is the proper.  who knew. not important, but if you wanna.

File: lib/secureAccessMethods.js [hunk 5]
```diff
@@ -174,10 +174,28 @@ function createSecure(user, options) {
...
+ */
+function bypass(method, options) {
+    const self = this;
+    const shield = self.constructor.shield;
+
+    if (!_.includes(shield.protectedMethods, method)) {
+        return Promise.reject(new Error('no such protected method'));
+    }
+    const protectedMethod = `_${method}`;
+    return shield[protectedMethod].call(self, options);
```
Comment (cdaringe) [inline lib/secureAccessMethods.js:191]: no action req'd, but `self` isn't really needed in here, from what i can see

File: package.json [hunk 6]
```diff
@@ -2,12 +2,12 @@
   "name": "bookshelf-shield",
   "version": "2.1.2",
   "description": "Access control list based authorization for bookshelf models",
-  "main": "dist/main.js",
+  "main": "lib/main.js",
```
Comment (cdaringe) [inline package.json:5]: 👍 :)

File: package.json [hunk 7]
```diff
@@ -23,6 +23,7 @@
     "grunt-contrib-jshint": "0.11.0",
     "grunt-jscs": "1.5.0",
     "grunt-mocha-cli": "1.12.0",
+    "istanbul": "^0.4.3",
```
Comment (cdaringe) [inline package.json]: use it or lose it, BRO

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (rssk) [inline .github/PULL_REQUEST_TEMPLATE.md]: Has anyone really been far even as decided to use even go want to do look more like?

File: lib/secureAccessMethods.js (hunk 4 above)
Comment (rssk) [inline lib/secureAccessMethods.js:181]: Huh, this was generated by the atom plugin, maybe a PR is in order?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (cdaringe) [inline .github/PULL_REQUEST_TEMPLATE.md]: huh?

File: lib/secureAccessMethods.js (hunk 4 above)
Comment (cdaringe) [inline lib/secureAccessMethods.js:181]: really, it got the return type??

File: package.json (hunk 7 above)
Comment (rssk) [inline package.json]: I was gonna wait till the eslint + newest coinsvalidate, not sure actually why it was installed (I didnt add it)... I think it was coinsvalidate's fault as well... :-P

File: lib/secureAccessMethods.js (hunk 5 above)
Comment (rssk) [inline lib/secureAccessMethods.js:191]: I just copied Dylan's calls above ie: `return shield._fetchAll.call(self, options)...`, but you're probably right

File: package.json (hunk 7 above)
Comment (cdaringe) [inline package.json]: still here!  should scrap it.  don't you go on blamin' poor ol' c-v for being a hard workin' soul...

**Pull request merged**